core/src/main/java/com/klaviyo/coresdk/networking/requests/KlaviyoRequest.kt

```diff
@@ -5,14 +5,19 @@ import com.klaviyo.coresdk.networking.NetworkBatcher
 internal abstract class KlaviyoRequest: NetworkRequest() {
     companion object {
         internal const val BASE_URL = "https://a.klaviyo.com"
+
+        internal const val ANON_KEY = "\$anonymous"
     }
 
     override var queryData: String? = null
     override var payload: String? = null
 
+    internal abstract fun addAnonymousIdToProps()
```

Comment:
Track and identify requests insert anonymous IDs into different property maps, so I abstracted this function for each

Reply:
you can sorta flip this dependence, this class knows the key and the value for anonymous is always the same no matter the type of request, so you can do
```
internal fun addAnonymous(map: Map<String, String>): Map<String, String> {
    return map.plus(ANON_KEY to "Android:${ConfigFileUtils.readOrCreateUUID()}")
}
```

and in the `buildKlaviyoJsonQuery` use this to modify it

Reply:
I had a similar idea but didn't do that cause I didn't want to bloat the single-purpose of `buildKlaviyoJsonQuery`. I wanted that function to just take the existing data and simply convert it to a `JSONObject` and not worry about what other data gets filled in later.
This is nicer for adding requests in the future though. And its not a massive refactor for me to come back later and redefine some of this logic if I feel I'm bloating an internal function or two

Reply:
Also just a very minor point, I didn't use `plus` to add to maps cause it rebuilds the entire map each time and I felt it may be more effective to use `put` or `set` in a mutable map instead.